controllers/front/CartController.php

```diff
@@ -328,10 +352,18 @@ protected function processChangeProductInCart()
         // Check product quantity availability
         if ($this->id_product_attribute) {
             if (!Product::isAvailableWhenOutOfStock($product->out_of_stock) && !Attribute::checkAttributeQty($this->id_product_attribute, $qty_to_check)) {
+                if ($isUpdateOperationDown) {
+                    $this->updateOperationError = $this->trans('There are not enough products in stock', array(), 'Shop.Notifications.Error');
+                } else {
+                    $this->errors[] = $this->trans('There are not enough products in stock', array(), 'Shop.Notifications.Error');
+                }
+            }
+        } elseif (!$product->checkQty($qty_to_check && !$isUpdateOperationDown)) {
+            if ($isUpdateOperationDown) {
+                $this->updateOperationError = $this->trans('There are not enough products in stock', array(), 'Shop.Notifications.Error');
+            } else {
```

Comment:
- Too many if/elseif/else. Please simplify this part.
- You should factorize the message creation (there should be only one occurrence of `$this->trans('There are not enough products in stock', array(), 'Shop.Notifications.Error')`